### PR TITLE
Respond with 400 instead of 500 when CSTG request validation fails

### PR DESCRIPTION
**Pull request by caroline-ttd**

### CODE REVIEW

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 1]
```diff
@@ -453,6 +453,10 @@ else if(emailHash != null) {
             input = InputUtil.normalizePhoneHash(phoneHash);
         }
 
+        if (this.phoneSupport ? !checkTokenInputV1(input, rc) : !checkTokenInput(input, rc)) {
```
Comment (jon8787) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:456]: This expression `this.phoneSupport ? !checkTokenInputV1(input, rc) : !checkTokenInput(input, rc)` appears 6 times in this file and ideally should be wrapped in a method

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java [hunk 2]
```diff
@@ -4162,6 +4162,56 @@ void cstgNoActiveKey(Vertx vertx, VertxTestContext testContext) throws NoSuchAlg
...
+                testContext,
+                respJson -> {
+                    assertFalse(respJson.containsKey("body"));
+                    assertEquals("Invalid Identifier", respJson.getString("message"));
+                    testContext.completeNow();
+                });
+    }
+
+    @Test
+    void cstgInvalidPhoneHashInput(Vertx vertx, VertxTestContext testContext) throws NoSuchAlgorithmException, InvalidKeyException {
```
Comment (jon8787) [inline src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java:4191]: possible to eliminate some of the duplication? the tests are identical other than the input. Can the input be parameterized?

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java (hunk 1 above)
Comment (mcollins-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:456]: The only difference between `checkTokenInputV1` and `checkTokenInput` is the error message returned.

Can we remove `checkTokenInputV1` and do the `this.phoneSupport` check in `checkTokenInput`?

Comment (caroline-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: I removed the `checkTokenInputV1` and did the `this.phoneSuppor`t check in `checkTokenInput`. Also invert the method.

File: src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java (hunk 2 above)
Comment (caroline-ttd) [reply src/test/java/com/uid2/operator/UIDOperatorVerticleTest.java]: That is a good idea, I used the `@ParameterizedTest`

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 3]
```diff
@@ -453,6 +453,10 @@ else if(emailHash != null) {
             input = InputUtil.normalizePhoneHash(phoneHash);
         }
 
+        if (checkTokenInput(input, rc)) {
```
Comment (jon8787) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:456]: `if (checkTokenInput)` makes it seem like it succeeded. How about we rename this method to `checkForInvalidTokenInput`?

**Review by jon8787 — CHANGES_REQUESTED** [review]

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java (hunk 3 above)
Comment (caroline-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: I rename checkTokenInput to checkForInvalidTokenInput. true for invalid input, false for valid input.

**Review by jon8787 — APPROVED** [review]

**Pull request merged**